### PR TITLE
Add storageClassName and volumeMode to datavolumes

### LINKED ISSUE #12: kubevirt_vm: storageClassName seems to be ignored
##### SUMMARY

The attribute "storageClassName" seems to be ignored by kubevirt_vm

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

kubevirt_vm

##### ANSIBLE VERSION

```paste below
ansible 2.9.14
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dominik/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.9 (default, Aug 19 2020, 17:05:11) [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]

```

##### CONFIGURATION

```paste below

```
empty by intention

##### OS / ENVIRONMENT

Fedora release 31 (Thirty One)

##### STEPS TO REPRODUCE

```yaml
[dominik@t460p ~]$ ansible-playbook -v "gdrive/CNV-Ceph HC/yml/02_vm1.yml"
Using /etc/ansible/ansible.cfg as config file
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match
'all'

PLAY [Playbook 2] *********************************************************************************************************

TASK [Gathering Facts] ****************************************************************************************************
ok: [localhost]

TASK [fedora11] ***********************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Timeout occurred while waiting for virtual machine to achieve 'running' state. Maybe try a higher wait_timeout value?"}

PLAY RECAP ****************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   


[dominik@t460p ~]$ cat "gdrive/CNV-Ceph HC/yml/02_vm1.yml"
---
# ansible-galaxy collection install community.general
# sudo dnf install python3-kubernetes python3-openshift
# https://github.com/kubevirt/ansible-kubevirt-modules/blob/master/examples/blog/part1/02_vm1.yaml
# https://docs.ansible.com/ansible/latest/collections/community/general/kubevirt_vm_module.html
- name: Playbook 2
  hosts: localhost
  connection: local
  tasks:
    - name: fedora11
      kubevirt_vm:
          state: running
          wait: true
          name: fedora11
          namespace: default
          memory: 1024M
          datavolumes:
            - name: rootdisk-dv
              source:
                http:
                  url: >-
                    http://ftp-stud.hs-esslingen.de/pub/fedora/linux/releases/33/Cloud/x86_64/images/Fedora-Cloud-Base-33-1.2.x86_64.qcow2
              pvc:
                # is this ignored?
                storageClassName: ocs-storagecluster-ceph-rbd
                accessModes:
                  - ReadWriteOnce
                storage: 5Gi
          disks:
            - name: rootdisk
              volume:
                dataVolume:
                  name: rootdisk-dv


[dominik@t460p ~]$ ansible --version
ansible 2.9.14
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/dominik/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.9 (default, Aug 19 2020, 17:05:11) [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]
[dominik@t460p ~]$ ansible-config dump --only-changed
[dominik@t460p ~]$ cat /etc/redhat-release 
Fedora release 31 (Thirty One)

```

##### EXPECTED RESULTS

```paste below
spec:
  dataVolumeTemplates:
  - apiVersion: cdi.kubevirt.io/v1alpha1
    kind: DataVolume
...
    spec:
      pvc:
...
        storageClassName: ocs-storagecluster-ceph-rbd
...
```

##### ACTUAL RESULTS

```paste below
[dominik@t460p ~]$ tmp/oc get vm/fedora11 -o  yaml
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachine
metadata:
  annotations:
    kubevirt.io/latest-observed-api-version: v1alpha3
    kubevirt.io/storage-observed-api-version: v1alpha3
  creationTimestamp: "2020-12-07T19:45:03Z"
  generation: 1
  managedFields:
  - apiVersion: kubevirt.io/v1alpha3
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        .: {}
        f:dataVolumeTemplates: {}
        f:running: {}
        f:template:
          .: {}
          f:metadata:
            .: {}
            f:labels:
              .: {}
              f:vm.cnv.io/name: {}
          f:spec:
            .: {}
            f:domain:
              .: {}
              f:devices:
                .: {}
                f:disks: {}
              f:resources:
                .: {}
                f:requests:
                  .: {}
                  f:memory: {}
            f:volumes: {}
    manager: OpenAPI-Generator
    operation: Update
    time: "2020-12-07T19:45:03Z"
  - apiVersion: kubevirt.io/v1alpha3
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubevirt.io/latest-observed-api-version: {}
          f:kubevirt.io/storage-observed-api-version: {}
      f:status: {}
    manager: virt-controller
    operation: Update
    time: "2020-12-07T19:45:03Z"
  name: fedora11
  namespace: default
  resourceVersion: "3470747"
  selfLink: /apis/kubevirt.io/v1alpha3/namespaces/default/virtualmachines/fedora11
  uid: 9d02eab9-0d36-4d7b-bdf5-1da08d1abeae
spec:
  dataVolumeTemplates:
  - metadata:
      creationTimestamp: null
      name: rootdisk-dv
    spec:
      pvc:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 5Gi
      source:
        http:
          url: http://ftp-stud.hs-esslingen.de/pub/fedora/linux/releases/33/Cloud/x86_64/images/Fedora-Cloud-Base-33-1.2.x86_64.qcow2
  running: true
  template:
    metadata:
      creationTimestamp: null
      labels:
        vm.cnv.io/name: fedora11
    spec:
      domain:
        devices:
          disks:
          - name: rootdisk
          - disk:
              bus: virtio
            name: rootdisk-dv
        machine:
          type: pc-q35-rhel8.2.0
        resources:
          requests:
            memory: 1024M
      volumes:
      - dataVolume:
          name: rootdisk-dv
        name: rootdisk
      - dataVolume:
          name: rootdisk-dv
        name: rootdisk-dv

```

### PR DESCRIPTION
**Pull request by snecklifter**
##### SUMMARY
Currently the code ignores additional parameters and both
storageClassName and volumeMode are commonly used. This commit
adds them.

Fixes #12

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
kubevirt_vm

##### ADDITIONAL INFORMATION
N/A

### CODE REVIEW

**Comment by Akasurde** [comment]
@snecklifter Thanks for the contribution. The CI failure is not related to this PR.

File: plugins/module_utils/kubevirt.py [hunk 1]
```diff
@@ -180,6 +180,8 @@ def _define_datavolumes(self, datavolumes, spec):
             dvt['metadata']['name'] = dv.get('name')
             dvt['spec']['pvc'] = {
                 'accessModes': dv.get('pvc').get('accessModes'),
+                'storageClassName': dv.get('pvc').get('storageClassName'),
```
Comment (Akasurde) [inline plugins/module_utils/kubevirt.py:183]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by Andersson007** [comment]
cc @tima as a maintainer could you please take a look?

cc @machacekondra @mmazur @pkliczewski you were in the team_kubevirt in ansible/ansible before this stuff was moved to this repo. Are you interested in becoming maintainers of this repository?
We can help with figuring out how to maintain this collection if needed.

The failing check is unrelated, please ignore.

**Comment by Andersson007** [comment]
@snecklifter thanks for rebasing! I pinged folks internally as the team_kubevirt members do not respond...

**Comment by Andersson007** [comment]
@Akasurde the changelog has been added. If you know anyone who can review this and the other PR, please ping them:)

**Comment by Andersson007** [comment]
@machacekondra @mmazur @pkliczewski could you please review this PR?

File: changelogs/fragments/add-missing-datavol-params.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - kubevirt_vm - Adds missing storageClassName and volumeMode parameters to datavol class (https://github.com/ansible-collections/community.kubevirt/issues/12)
```
Comment (alinabuzachis) [inline changelogs/fragments/add-missing-datavol-params.yaml]:
```suggestion
  - kubevirt_vm - Adds missing ``storageClassName`` and ``volumeMode`` parameters to datavol class (https://github.com/ansible-collections/community.kubevirt/issues/12)
```

**Review by alinabuzachis — APPROVED** [review]
@snecklifter Thank you. I left one comment with minor neats. This LGTM!

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@snecklifter thanks for the contribution! @alinabuzachis @Akasurde thanks for reviewing!

**Pull request merged**